pages/nft/iscn/_iscnId.vue

```diff
@@ -912,39 +926,58 @@ export default class NFTTestMintPage extends Vue {
     return null;
   }
 
-  async mintNFT() {
-    await this.initIfNecessary()
-    try {
-      if (!this.signer) return
-      const signingClient = await getSigningClient()
-      await signingClient.setSigner(this.signer)
-
-      if (!this.nftsIds.length) this.nftsIds = [...Array(this.premintAmount).keys()]
+  getNFTMessages(classId = this.classId) {
+    if (!this.nftsIds.length) this.nftsIds = [...Array(this.premintAmount).keys()]
         .map((_) => `${this.isWritingNFT ? 'writing-' : ''}${uuidv4()}`);
       const nfts = this.nftsIds.map(id => this.getMintNftPayload(id))
       const mintMessages = nfts.map((i) =>
-        formatMsgMintNFT(this.address, this.classId, i),
+        formatMsgMintNFT(this.address, classId, i),
       )
       const nftsToSend = nfts.filter((_, index) => index >= this.reserveNft)
       const sendMessages = nftsToSend.map((i) =>
-        formatMsgSend(this.address, LIKER_NFT_API_WALLET, this.classId, i.id),
+        formatMsgSend(this.address, LIKER_NFT_API_WALLET, classId, i.id),
       )
+    return {
+      mintMessages,
+      sendMessages,
+    }
+  }
+
+  async mintNFT() {
+    await this.initIfNecessary()
+    try {
+      if (!this.signer) return
+      const signingClient = await getSigningClient()
+      await signingClient.setSigner(this.signer)
+
       logTrackerEvent(this, 'IscnMintNFT', 'MintNFT', this.classId, 1);
 
       const shouldMintNFT = !this.mintNFTResult
       const shouldSendNFT = this.isWritingNFT && !this.sendNFTResult
 
+      const { mintMessages, sendMessages } = this.getNFTMessages();
+
+      let messages: any[] = []
       if (shouldMintNFT) {
-        this.mintNFTResult = await signingClient.sendMessages(
-          this.address,
-          mintMessages,
-        )
+        messages = messages.concat(mintMessages)
+        if (this.isTransactionSizeLimited) {
+          this.mintNFTResult = await signingClient.sendMessages(
+            this.address,
+            messages,
+          )
+          messages = [];
+        }
```

Comment:
No need to reset `message` in this case?
```suggestion
        if (this.isTransactionSizeLimited) {
          this.mintNFTResult = await signingClient.sendMessages(
            this.address,
            mintMessages,
          )
        } else {
          messages.push(...mintMessages)
        }
```